Lib/asyncio/events.py

```diff
@@ -101,7 +101,8 @@ class TimerHandle(Handle):
     __slots__ = ['_scheduled', '_when']
 
     def __init__(self, when, callback, args, loop, context=None):
-        assert when is not None
+        if when is None:
```

Comment:
TimerHandle is never instantiated explicitly; the check should be moved into loop.call_later()/loop.call_at() methods

Reply:
In tests, it is tested by instantiating  directly, doing this would require removing the test.

Reply:
I think, removing the test and adding new ones for call_later()/call_at() is what we should do.

Reply:
Done